core/admin/mailu/internal/views/autoconfig.py

```diff
@@ -0,0 +1,163 @@
+from mailu.internal import internal
+
+from flask import current_app as app
+import flask
+
+@internal.route("/autoconfig/mozilla")
+def autoconfig_mozilla():
+    # https://wiki.mozilla.org/Thunderbird:Autoconfiguration:ConfigFileFormat
+    hostname = app.config['HOSTNAME']
+    xml = f'''<?xml version=\"1.0\"?>
+<clientConfig version=\"1.1\">
+<emailProvider id=\"%EMAILDOMAIN%\">
+<domain>%EMAILDOMAIN%</domain>
+
+<displayName>Email</displayName>
+<displayShortName>Email</displayShortName>
+
+<incomingServer type=\"imap\">
+<hostname>{hostname}</hostname>
+<port>993</port>
+<socketType>SSL</socketType>
+<username>%EMAILADDRESS%</username>
+<authentication>password-cleartext</authentication>
+</incomingServer>
+
+<outgoingServer type=\"smtp\">
+<hostname>{hostname}</hostname>
+<port>465</port>
+<socketType>SSL</socketType>
+<username>%EMAILADDRESS%</username>
+<authentication>password-cleartext</authentication>
+<addThisServer>true</addThisServer>
+<useGlobalPreferredServer>true</useGlobalPreferredServer>
+</outgoingServer>
+
+<documentation url=\"https://{hostname}/admin/client\">
+<descr lang=\"en\">Configure your email client</descr>
+</documentation>
+</emailProvider>
+</clientConfig>\r\n'''
+    return flask.Response(xml, mimetype='text/xml', status=200)
+
+@internal.route("/autoconfig/microsoft", methods=['GET', 'POST'])
+def autoconfig_microsoft():
+    # https://docs.microsoft.com/en-us/previous-versions/office/office-2010/cc511507(v=office.14)?redirectedfrom=MSDN#Anchor_3
+    hostname = app.config['HOSTNAME']
+    xml = f'''<?xml version=\"1.0\" encoding=\"utf-8\" ?>
+<Autodiscover xmlns=\"https://schemas.microsoft.com/exchange/autodiscover/responseschema/2006\">
```

Comment:
My hunch is that the schema must be in HTTP and not HTTPS
 xmlns="http://schemas.microsoft.com/exchange/autodiscover/responseschema/2006"

Yes, the XSD agrees with me. See https://docs.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxdscli/9c9284b4-244d-4692-93a0-f5c44bf153ad

Reply:
Doh; it was searched and replaced naively: 23a8943b